test/behavior/int_comparison_elision.zig

```diff
@@ -0,0 +1,75 @@
+const std = @import("std");
+const minInt = std.math.minInt;
+const maxInt = std.math.maxInt;
+const builtin = @import("builtin");
+
+test "int comparison elision" {
+    testIntEdges(u0);
+    testIntEdges(i0);
+    testIntEdges(u1);
+    testIntEdges(i1);
+    testIntEdges(u4);
+    testIntEdges(i4);
+
+    // TODO: support int types > 128 bits wide in other backends
+    if (builtin.zig_backend == .stage2_wasm) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_c) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_x86_64) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_aarch64) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_arm) return error.SkipZigTest; // TODO
+
+    // TODO: panic: integer overflow with int types > 65528 bits wide
+    // TODO: LLVM generates too many parameters for wasmtime when splitting up int > 64000 bits wide
+    testIntEdges(u64000);
+    testIntEdges(i64000);
+}
+
+fn testIntEdges(comptime T: type) void {
+    var runtime_val: T = undefined;
+    const undef_const: T = undefined;
+    testIntEdgesInner(T, runtime_val);
+    testIntEdgesInner(T, undef_const);
```

Comment:
Both of these test the same thing since the comptime known value isn't visible to the inner function.

Reply:
Ok, I see the problem. The distinction between variable and `Value.undef` is lost because from the perspective of the inner function, they are both runtime variables.

I would still like to test both cases, so I will rewrite this a different way.

Reply:
Would making `testIntEdgesInner` an `inline fn` fix this? Or does inlining only affect generated machine code, not semantic analysis?

Reply:
It would make no difference without #7772, you're better off manually inlining the test.

Reply:
While I'm getting this to work rebased on latest master: Would an `inline for` capture have the same limitation? If so I will just copy/paste those test statements for both cases, no biggie.

Reply:
It shouldn't if you use a tuple.

Reply:
I love the tuple idea, but unfortunately this produces:
```
error: TODO implement airAggregateInit for target.Target.Cpu.Arch.aarch64
```

So I can just do the ugly duplicated code for now so that all platforms get coverage.

Reply:
It looks like you are disabling the test for the AArch64 backend. I suspect you are getting this error from a different test case that is missing the line to disable it for this backend:

```zig
if (builtin.zig_backend == .stage2_aarch64) return error.SkipZigTest; // TODO
```

Reply:
I should have pasted the full output, it's definitely this file:

```
test/behavior/int_comparison_elision.zig:29:1: error: TODO implement airAggregateInit for target.Target.Cpu.Arch.aarch64
fn testIntEdges(comptime T: type) void {
```

It only skips targets before calling `testIntEdges` with the large int sizes -- it's still running tests with smaller sizes before that. Regardless, not using the tuple approach lets the test build on all backends (with smaller int sizes).